### PR TITLE
⚠️  Promote supported K8s versions to 1.23 - 1.25

### PR DESCRIPTION
**Pull request by roytman**
Promote supported K8s versions to 1.23 - 1.25

/fixes https://github.com/fybrik/fybrik/issues/1900 

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

### CODE REVIEW

File: site/docs/contribute/environment.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ make install-tools
 This installs additional dependencies to `hack/tools/bin`. The `make` targets (e.g., `make test`) are configured to use the binaries from `hack/tools/bin`. However, you may want to add some of these tools to your system PATH for direct usage from your terminal (e.g., for using `kubectl`).
 
 Please note: For fybrik version 0.5.x and lower, Helm version greater than 3.3 but less than 3.7 is required when contributing. 
-On the other side, for fybrik v0.6.x, Helm v3.7 or above is required. 
+On the other side, for fybrik v0.6.x, Helm v3.7 or above is required. Fybrik v1.2 requires Helm 1.10
```
Comment (shlomitk1) [inline site/docs/contribute/environment.md:33]: 3.10

**Review by shlomitk1 — CHANGES_REQUESTED** [review]
Helm 3.10 instead of 1.10

File: site/docs/contribute/environment.md (hunk 1 above)
Comment (roytman) [reply site/docs/contribute/environment.md]: fixed

**Comment by roytman** [comment]
> Helm 3.10 instead of 1.10

done

**Review by revit13 — DISMISSED** [review]

**Review by shlomitk1 — APPROVED** [review]

**Pull request merged**